src/ai/backend/manager/background.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+from asyncio import AbstractEventLoop, Task
+import functools
+import logging
+from typing import Callable, Set, Union
+import uuid
+
+from aiojobs import Scheduler
+
+from ai.backend.common.logging import BraceStyleAdapter
+
+from ..gateway.events import EventDispatcher
+
+log = BraceStyleAdapter(logging.getLogger('ai.backend.gateway.stream'))
+
+
+class ProgressReporter:
+    event_dispatcher: EventDispatcher
+    total_progress: Union[int, float]
+    current_progress: Union[int, float]
+    task_id: uuid.UUID
+    loop: AbstractEventLoop
+
+    def __init__(self, event_dispatcher: EventDispatcher,
+                       task_id: uuid.UUID,
+                       loop: AbstractEventLoop):
+        self.event_dispatcher = event_dispatcher
+        self.task_id = task_id
+        self.loop = loop
+
+    async def set_progress_total(self, value: Union[int, float]):
+        self.total_progress = value
+
+    async def update_progress(self, current: Union[int, float], message: str = None):
+        self.current_progress = current
+        await self.event_dispatcher.produce_event(
+            'task_update',
+            (str(self.task_id), current, self.total_progress, message, )
+        )
+
+
+class BackgroundTask:
```

Comment:
This name may mislead the readers to think that the class represents individual background tasks.
Let's rename to `BackgroundTaskManager`.